src-common/main/scala/uk/gov/hmrc/play/audit/http/connector/AuditCounter.scala

```diff
@@ -64,7 +75,7 @@ private[connector] trait UnpublishedAuditCounter extends AuditCounter {
       )
       publishedSequence.set(currentSequence)
       if (isFinal) {
-        finalSequence.set(true)
+        finalSequence.set(Some(currentSequence))
```

Comment:
The only way I can imagine this going wrong is if there are two concurrent services shutting down, publishing to the same path. 

I _think_ the platform provides this invariant - that each host name runs one host only.  Have you checked if this is the case on future platform?